docs_src/design/ucr/Global Common Configuration.md

```diff
@@ -0,0 +1,169 @@
+        ...
+        # Common Security Service Metrics
+        SecuritySecretsRequested = false
+        SecuritySecretsStored = false
+        SecurityConsulTokensRequested = false
+        SecurityConsulTokenDuration = false
+        [Writable.Telemetry.Tags] # Contains the service level tags to be attached to all the service's metrics
+    #    Gateway="my-iot-gateway" # Tag must be added here or via Consul Env Override can only chnage existing value, not added new ones.
+    ...
+    [MessageQueue]
+    Protocol = "redis"
+    Host = "localhost" <override in compose file same for every service>
+    Port = 6379
+    Type = "redis"
+    AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+    SecretName = "redisdb"
+    PublishTopicPrefix = <Service Specific>
+    SubscribeEnabled = <Service Specific>
+    SubscribeTopic = <Service Specific>
+      [MessageQueue.Topics]
+      <service specific name> = <Service specific value>
+      ...
+      [MessageQueue.Optional]
+      # Default MQTT Specific options that need to be here to enable evnironment variable overrides of them
+      ClientId = <Service Specific>
+      Qos =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+      KeepAlive = "10" # Seconds (must be 2 or greater)
+      Retained = "false"
+      AutoReconnect = "true"
+      ConnectTimeout = "5" # Seconds
+      SkipCertVerify = "false"
+      # Additional Default NATS Specific options that need to be here to enable evnironment variable overrides of them
+      Format = "nats"
+      RetryOnFailedConnect = "true"
+      QueueGroup = ""
+      Durable = ""
+      AutoProvision = "true"
+      Deliver = "new"
+      DefaultPubRetryAttempts = "2"
+      Subject = "edgex/#" # Required for NATS Jetstram only for stream autoprovsioning
+    ```
+In the above example only the **PublishTopicPrefix**, **SubscribeTopic**, **SubscribeEnabled**, **MessageQueue.Topics** and **ClientId** settings have unique values to that of the default EdgeX deployment values. 
+
+!!! note
+    In Levski release App Services don't have the **MessageQueue** section and Core Command's is **MessageQueue.Internal**. These inconstancies will be rectified in EdgeX 3.0 so all EdgeX services have the same **MessageQueue** section specified in the same manner. Also in EdgeX 3.0, the **PublishTopicPrefix** and **SubscribeTopic** settings will be replaced by entries in **MessageQueue.Topics**.
+
+There are other similar common sections not shown shown above. As can be seen from the two examples above there is much duplication of configuration settings across all the EdgeX services. This gives rise to the need to have all these common duplicate configuration settings in a single global source. 
+
+### Existing solutions
+There are no existing solutions for global configuration for EdgeX since the current configuration implementation is specific to EdgeX.

```

Comment:
Currently, when a Configuration Provider is used, can't a script be used to set and change all these values so that services load them automatically (writable fields) or after a reboot?

Reply:
Yes, but that isn't an existing solution now.

Reply:
True, it isn't an existing built-in solution. But this section of the UCR could go  beyond the EdgeX reference implementation and cover how would someone do it right now, if they wanted to.

The description from the template is: 
> ### Existing solutions
> How is the given use case currently implemented in the industry? List and describe each approach. Highlight possible gaps.

The purpose of writing that is to help the community and contributors identify how/if this use case can be realized (given the EdgeX gaps), what are EdgeX gaps, and how important it is to have it added to EdgeX.

Reply:
IMO, in this UCR "Existing Solutions"  refers to "How is global configuration" is implemented in other solutions, not how EdgeX current implements configuration. That said I have reworded this slightly and add a reference to the previous ADR for service configuration.